openstack/datadog_checks/openstack/openstack.py

```diff
@@ -945,9 +919,37 @@ def get_all_servers(self, i_key, collect_all_tenants, filter_by_host=None):
                 except KeyError as e:
                     self.log.debug("Server: %s has already been removed from the cache", new_server['server_id'])
 
-        return self.server_details_by_id
+    def filter_excluded_servers(self):
+        proj_list = set([])
+        servers_copy = copy.deepcopy(self.server_details_by_id)
+        if self.exclude_server_id_rules:
+            # Filter out excluded servers
+            for exclude_id_rule in self.exclude_server_id_rules:
+                for server_id in self.server_details_by_id.keys():
+                    if re.match(exclude_id_rule, server_id):
+                        del self.server_details_by_id[server_id]
```

Comment:
Perhaps there is a better way to do this, but I was only creating the copy so that I could iterate through it and delete items from the original self.server_details_by_id list. That way the class object has the true set of servers it will have to make calls for later.

Reply:
let's keep it as is then